frontend/components/app.jsx

```diff
@@ -1,32 +1,54 @@
 import React from 'react';
 import { Route, Link, NavLink, IndexRoute } from 'react-router-dom';
+import io from 'socket.io-client';
 
 import Nav from './nav';
 
 import Home from './home';
 import Lobby from './lobby';
 import Main from './main';
+import Puzzle from './puzzle';
 import Puzzle1 from './puzzle1';
 import Puzzle2 from './puzzle2';
 import Puzzle3 from './puzzle3';
 import Puzzle4 from './puzzle4';
 
-class App extends React.Component {
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    // TODO: update URL
+    const socket = io('http://localhost:8000');
 
-  render() {
+    socket.on('game_state_update', (data) => {
+      this.setState({ gameState: data });
+    });
+
+    this.state = {
+      gameState: {},
+      socket: socket,
+      send: this.send.bind(this),
+    };
+  }
 
+  send(eventName, data) {
+    console.log(data);
+    const { socket } = this.state;
+    socket.emit(eventName, data);
```

Comment:
so we want to call this to submit an answer to the server within the react/javascript function on the puzzle's submit button? The end thing should be socket.emit('submit', {puzzle2:'see o double you'})?